### ISSUE TITLE
Feature Request: Dark Mode

### ISSUE THREAD

**Issue by septatrix** [AliasIO/Wappalyzer#2776]
Description
-

**Do you want to request a *feature*, a *new application detection* or report a *bug*?**
Feature

**Is your issue about WebExtension driver (Chrome & Firefox), the website, the NPM driver or the bookmarklet ?**
Web extension

**What is the current behavior ?**
The extension popup has a light theme regardless of the browser setting.

**What is the expected behavior ?**
The extension popup switches to a dark theme when the browser uses one.

**Which versions of Wappalyzer, and which browser / OS are affected by this issue ? Did this work in previous versions ?**
This affects all versions of Wappalyzer running in browser supporting the css `prefers-color-scheme`-media query.

Reasoning
-

Chrome and Firefox now both support the corresponding ability to query a users color preference and allow one to change the color scheme based on this. I would like to implement this functionality into the extensions popup window as many website can be styled with other extensions but extensions themself can't be styled by 3rd party extensions.

I would be willing to work in this issue in my free time if this feature is of interest
-

I can't guarantee any deadline though I'd like to look into the technical details of implementing such a dark mode.

Possible Obstacles
-

As already mentioned in #2757 some icons are poorly visible on a dark background. This though doesn't only applies to a potential dark mode but also the preview in the extension tray. It would then however also extend to the popup area.
There are three possible ways to solve this problem i thought of:
1. Add a white background to the icons in the popup
2. Provide alternative icons for problematic logos - these then need to be chosen dynamically
3. Fix the problematic logos by e.g. adding a corresponding background or outline

Personally I think option 3 would be easy to implement and the best for the long run. This would also resolve issue #2757 in the hotbar.
At first i would try to implement the first option because it would not involve going through all ~1000 icons. If possible we could also try to draw the icon on a white canvas and use that as a toolbar icon to resolve the above issue. This however seems like a hack and fixing the symptoms instead of the cause.

I would like to hear your feedback on this... :D

**Comment by Septatrix** [comment]
A demo of how such a dark mode may look like:

![Screenshot from 2019-09-15 03-17-47](https://user-images.githubusercontent.com/24257556/64915458-8d13d000-d767-11e9-8d50-61693337c438.png)

The background color is `#333` which seems to fit chrome's native dark color pretty well...
As one can see, the GitHub Pages and jQuery icons don't have a lot of contrast on the background. Additionally the pin icon when hovering (top left category) has a fixed color. A possible solution would be to use a styleable svg or create two png files. Also the underline and version badge (under and next to Bootstrap) have been styled.

To replicate inspect the popup and change the stylesheet to the following:

[popup.css.txt](https://github.com/AliasIO/Wappalyzer/files/3613313/popup.css.txt)

**Comment by kyletaylored** [comment]
I went ahead and added some small support for Dark mode using the `prefers-color-scheme` media query. Let me know what you think!

![Screenshot](https://user-images.githubusercontent.com/1759794/65552021-e1961880-dee8-11e9-9ce2-a4f66da33938.png)

**Comment by Septatrix** [comment]
Yeah that looks really good. Also stumbled upon `drop-shadow` which indeed gives good results for alle icons I tested so far - however I didn't bother posting because I got no response yet if a dark mode is desired at all.

One problem which still exists but can easily be solved is the pin icon which we could change into a SVG, add a second variant or apply some CSS filters to make it white/lighter (which I also already tried - but can't show atm because I'm posting from my phone and it should be self explanatory).
If we would want to go one step further and address the icon contrast in the omnibar etc we could dynamic draw the image onto a canvas with a shadow similar to the case one and use it's image data as the new icon which should be possible according to the Extension(?) API.
